### PR TITLE
feat(365)-7: Add getScmContexts and canHandleWebhook functions

### PR DESCRIPTION
**Pull request by yokawara**
## Context
I'd like to implement the ability to use multiple scms feature (https://github.com/screwdriver-cd/screwdriver/issues/365).
The other related modules are also being modified.

## Objective
This PR adds `_getScmContexts` and `_canHandleWebhook` functions for the multiple scms feature.

## References
issue: https://github.com/screwdriver-cd/screwdriver/issues/365
Before merging for test success: https://github.com/screwdriver-cd/scm-base/pull/43
related changes:
- https://github.com/screwdriver-cd/data-schema/pull/152
- https://github.com/screwdriver-cd/data-schema/pull/155
- https://github.com/screwdriver-cd/screwdriver/pull/632
- https://github.com/screwdriver-cd/models/pull/180
- https://github.com/screwdriver-cd/scm-router/pull/1
- https://github.com/screwdriver-cd/scm-base/pull/43
- https://github.com/screwdriver-cd/scm-github/pull/65
- https://github.com/screwdriver-cd/scm-bitbucket/pull/41

### CODE REVIEW

**Comment by bdangit** [comment]
looks good so far @yokawara   let me know when its okay to merge.

File: index.js [hunk 1]
```diff
@@ -676,16 +691,24 @@ class GitlabScm extends Scm {
      * @return {Promise}
      */
     _getBellConfiguration() {
-        const bellConfig = {
+        const scmContexts = this._getScmContexts();
+        const scmContext = scmContexts[0];
+        const cookie = this.config.gitlabHost
+            ? `gitlab-${this.config.gitlabHost}`
+            : 'gitlab-gitlab.com';
```
Comment (catto) [inline index.js]: you can reduce use of a hardcode by defining the default host to an instance variable.

**Comment by yokawara** [comment]
@bdangit Now we can merge this PR. Please reconfirm our new changes, and merge them if there is no problem.

**Comment by bdangit** [comment]
Thanks @yokawara.  Later in the evening I'll get this pushed into Npm.

**Comment by bdangit** [comment]
hey @yokawara, sorry its taking me a bit longer.  I'm trying to remember what I did to test and publish (i wish i had setup an automated job for this).  I'm running into an error where the tests are failing on the following:

```
1) index parseUrl resolves to the ccorrect parsed url for ssh:
    Error: you must provide a joi schema

2) index parseUrl resolves to the correct parsed url for https:
    Error: you must provide a joi schema

3) index getScmCContexts returns a default sccmContext:
    Error: you must provide a joi schema

4) index getSccmContexts returns a scmContext for user setting gitlabHost:
    Error: you must provide a joi schema

```

I'll have to review the code again, unless you can quickly give me the answer :).

**Pull request merged**